### PR TITLE
Mesh support for STL files

### PR DESCRIPTION
**Pull request by bosonbaas**
With the current mesh tooling, STL formatted mesh files are imported with each triangle disconnected from its neighbors. 
This PR merges these meshes pointwise (points at the same place are merged into a single point) in the conversion to delta sets.

### CODE REVIEW

**Comment by bosonbaas** [comment]
This PR has also changed the graphing backend from AbstractPlotting to Makie. Apparently this transfer was made by the Makie community in their last version update.

File: src/MeshInterop.jl [hunk 1]
```diff
@@ -69,6 +76,10 @@ that it will not preserve any normal, texture, or other data beyond points and
 triangles from the mesh file.
 """
 function EmbeddedDeltaSet2D(fn::String)
-	EmbeddedDeltaSet2D(FileIO.load(fn))
+  if(fn[(end-2):end] == "stl")
```
Comment (epatters) [inline src/MeshInterop.jl:79]: Should use the `splitext` function to get the file extension.

File: src/MeshInterop.jl [hunk 2]
```diff
@@ -48,13 +48,20 @@ end
 This operator should work for any triangular mesh object. Note that it will
 not preserve any normal, texture, or other attributes from the Mesh object.
 """
-function EmbeddedDeltaSet2D(m::GeometryBasics.Mesh)
-  coords = coordinates(m)
+function EmbeddedDeltaSet2D(m::GeometryBasics.Mesh; force_unique=false)
+  coords = metafree.(coordinates(m))
+  ind_map = collect(1:length(coords))
```
Comment (epatters) [inline src/MeshInterop.jl:53]: I could be wrong, but I doubt you need a `collect` here.

**Review by epatters — COMMENTED** [review]
Thanks Andrew! Minor comments below.

File: src/MeshInterop.jl [hunk 3]
```diff
@@ -48,13 +48,20 @@ end
 This operator should work for any triangular mesh object. Note that it will
 not preserve any normal, texture, or other attributes from the Mesh object.
 """
-function EmbeddedDeltaSet2D(m::GeometryBasics.Mesh)
-  coords = coordinates(m)
+function EmbeddedDeltaSet2D(m::GeometryBasics.Mesh; force_unique=false)
```
Comment (epatters) [inline src/MeshInterop.jl:51]: Probably worth documenting what the keyword argument `force_unique` does and why you need it for STL files.

**Comment by bosonbaas** [comment]
Thanks for the feedback! Those should be taken care of in this latest commit.

**Comment by epatters** [comment]
Thanks! Merging.

**Pull request merged**